src/Approximations/decompositions.jl

```diff
@@ -28,61 +28,60 @@ function overapproximate(X::LazySet)::HPolygon
 end
 
 """
-    overapproximate(X, ɛ)
+    overapproximate(S::LazySet, ɛ::Float64)::HPolygon
 
 Return an ɛ-close approximation of the given 2D set (in terms of Hausdorff
 distance) as a polygon.
 
 ### Input
 
-- `X` -- lazy set, assumed to be two-dimensional
-- `ɛ` -- the error bound
+- `S` -- convex set, assumed to be two-dimensional
+- `ɛ` -- error bound
 
 ### Output
 
 A polygon in constraint representation.
 """
-function overapproximate(X::LazySet, ɛ::Float64)::HPolygon
-    constraints = [LinearConstraint(ci.d1, dot(ci.d1, ci.p1)) for ci in approximate(X, ɛ)]
+function overapproximate(S::LazySet, ɛ::Float64)::HPolygon
+    @assert dim(S) == 2
+
+    constraints =
+        [LinearConstraint(ci.d1, dot(ci.d1, ci.p1)) for ci in approximate(S, ɛ)]
     return HPolygon(constraints)
 end
 
 """
-    decompose(X)
+    decompose(S::LazySet)::CartesianProductArray
 
-Compute an overapproximation of the projections of the given set over each
-two-dimensional subspace using box directions.
+Compute an overapproximation of the projections of the given convex set over
+each two-dimensional subspace using box directions.
 
 ### Input
 
-- `X`  -- lazy set
+- `S` -- convex set
 
 ### Output
 
-A `CartesianProductArray` corresponding to the cartesian product of 2x2 polygons.
+A `CartesianProductArray` corresponding to the Cartesian product of
+``2 \\times 2`` box-shaped polygons.
 """
-function decompose(X::LazySet)::CartesianProductArray
-    n = LazySets.dim(X)
+function decompose(S::LazySet)::CartesianProductArray
+    n = dim(S)
     b = div(n, 2)
-
-    DIR_EAST_bi = sparsevec([], Float64[], n)  # TODO: use numeric eltype of X
-    DIR_NORTH_bi = sparsevec([], Float64[], n)
-    DIR_WEST_bi = sparsevec([], Float64[], n)
-    DIR_SOUTH_bi = sparsevec([], Float64[], n)
     result = Vector{HPolygon}(b)
 
     @inbounds for bi in 1:b
-        copy!(DIR_EAST_bi, sparsevec([2*bi-1], [1.], n))
-        pe_bi = dot(DIR_EAST, view(σ(DIR_EAST_bi, X), 2*bi-1:2*bi))
+        pe_bi = dot(DIR_EAST,
```

Comment:
yes, why not.

Reply:
The new version is actually more memory efficient (according to `@time`), probably because the sparse vector is still created. I did not understand the motivation for the `copy!`.